### DISCUSSION TITLE
Time consistency

### DISCUSSION THREAD

**Discussion by gtolarc** [keystonejs/keystone#5331] [Questions] [answered]
There is a need for a consistent policy regarding timezone of DateTime.

The DateTime format is internally managed as [fieldName]_utc, [fieldName]_offset, but I think delegating the timezone decision to the browser setting and removing the offset is a convenient way to reduce confusion.

And it would be better to change TextDayTimePicker/DayTimePicker in UI to just express it according to the timezone setting of the browser. How about this?

Timezone representation is a really important area in CMS, and current keystone5 is not clear about this, and there are bugs. I think it needs to be fixed quickly.

**Comment by gautamsi** [comment]
three is a field DateTimeUtc which does not have offset and it always saves in UTC.

**Reply by gtolarc** [reply]
I know. However, it doesn't seem intuitive. And it seems to be causing more confusion because there is a problem with the display of the datetime in the current keystone admin frontend.

**Reply by ChuckJonas** [reply, marked as answer]
The issue simply is the bug with the admin UI.  The database should store the time in UTC (as it does).  The time should be displayed in the local timezone of the user.

I think the UI component is just broken and selecting a date always overwrites your selection with the previous day.  

I created a bug displaying this behavior here:

https://github.com/keystonejs/keystone/issues/6115

**Reply by dcousens** [reply]
I think we can say this issue has been resolved by #6115, please open a different discussion if necessary

**Comment by bladey** [comment]
Thanks for the discussion @gtolarcm, @gautamsi and @ChuckJonas, we'll look into this in the near future.
